example/spec/kinesisTests/KinesisTestTriggerSpec.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const { s3 } = require('@cumulus/common/aws');
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 550000;
+
+const {
+  LambdaStep,
+  waitForCompletedExecution
+} = require('@cumulus/integration-tests');
+const { randomString } = require('@cumulus/common/test-utils');
+
+const { loadConfig } = require('../helpers/testUtils');
+const {
+  createOrUseTestStream,
+  putRecordOnStream,
+  waitForActiveStream,
+  waitForTestSfStarted
+} = require('../helpers/kinesisHelpers');
+
+const record = require('../../data/records/L2_HR_PIXC_product_0001-of-4154.json');
+
+const granuleId = record.product.name;
+const recordIdentifier = randomString();
+record.identifier = recordIdentifier;
+
+const testConfig = loadConfig();
+
+const lambdaStep = new LambdaStep();
+const streamName = testConfig.streamName;
+
+
+const recordFile = record.product.files[0];
+const expectedTranslatePayload = {
+  granules: [
+    {
+      granuleId: record.product.name,
+      files: [
+        {
+          path: 'cumulus-test-data/pdrs',
+          url_path: recordFile.uri,
+          bucket: record.bucket,
+          name: recordFile.name,
+          size: recordFile.size
+        }
+      ]
+    }
+  ]
+};
+
+const fileData = expectedTranslatePayload.granules[0].files[0];
+const filePrefix = `file-staging/${testConfig.stackName}/L2_HR_PIXC`;
+
+const fileDataWithFilename = {
+  ...fileData,
+  filename: `s3://${testConfig.buckets.private.name}/${filePrefix}/${recordFile.name}`,
+  bucket: testConfig.buckets.private.name,
+  url_path: '',
+  fileStagingDir: filePrefix
+};
+
+const expectedSyncGranulesPayload = {
+  granules: [
+    {
+      granuleId: granuleId,
+      files: [fileDataWithFilename]
+    }
+  ]
+};
+
+
+// When Cumulus is configured to trigger a CNM workflow from a Kinesis stream and a message appears on the stream, Cumulus triggers the workflow
+describe('The Cloud Notification Mechanism Kinesis workflow', () => {
+  const maxWaitTime = 1000 * 60 * 2;
+  let workflowExecution;
+  let executionStatus;
+  let s3FileHead;
+
+  afterAll(async () => {
+    await s3().deleteObject({
+      Bucket: testConfig.buckets.private.name,
+      Key: `${filePrefix}/${fileData.name}`
+    }).promise();
+  });
+
+  beforeAll(() => {
+    this.workflowExecution = null;
+  });
+
+  it('finds or creates a test stream.', async () => {
+    await createOrUseTestStream(streamName);
+  });
+
+  it('waits for the stream to be active.', async () => {
+    await waitForActiveStream(streamName);
+  });
+
+  it('places a record on the stream.', async () => {
+    await putRecordOnStream(streamName, record);
+  });
```

Comment:
I think these three tests `finds or creates a test stream`, `waits for the stream to be active` and `places a record on the stream` should be in a beforeAll block.